receptai.api/Repositories/RecipeVoteRepository.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.EntityFrameworkCore;
+using receptai.api.Dtos.RecipeVote;
+using receptai.api.Interfaces;
+using receptai.api.Mappers;
+using receptai.data;
+
+namespace receptai.api.Repositories;
+
+public class RecipeVoteRepository : IRecipeVoteRepository
+{
+    private readonly RecipePlatformDbContext _context;
+    private readonly IRecipeRepository _recipeRepository;
+
+    public RecipeVoteRepository(RecipePlatformDbContext context,
+        IRecipeRepository recipeRepository)
+    {
+        _context = context;
+        _recipeRepository = recipeRepository;
+    }
+
+    public async Task<RecipeVote?> GetByUserAndRecipeId(int userId, int recipeId)
+    {
+        var recipeVoteModel = await _context.RecipeVotes
+           .FirstOrDefaultAsync(rv => rv.UserId == userId &&
+               rv.RecipeId == recipeId);
+
+        return recipeVoteModel;
+    }
+
+    public async Task<List<RecipeVote>> GetByRecipeId(int recipeId)
+    {
+        var recipeVoteModels = await _context.RecipeVotes
+           .Where(rv => rv.RecipeId == recipeId)
+           .ToListAsync();
+
+        return recipeVoteModels;
+    }
+
+    public async Task<RecipeVote> CreateAsync(RecipeVote recipeVoteModel)
+    {
+        await _context.RecipeVotes.AddAsync(recipeVoteModel);
+        await _context.SaveChangesAsync();
+
+        await _recipeRepository.RecalculateVotesAsync(
+            recipeVoteModel.RecipeId);
```

Comment:
Wouldn't it make more sense to add new votes, then recalculate, then save changes?

Reply:
The change in list of votes must be saved first before getting it to recalculate votes, I think my solution is viable. Swapping them around does not work.